scanner.go

```diff
@@ -107,9 +112,17 @@ func (s *scanner) iterSingle(iter Scannable) (int, error) {
 	}
 
 	ptrs := generatePtrs(structFields)
-	scanOk := iter.Scan(ptrs...) // we only need to scan once
-	if !scanOk {
-		return 0, RowNotFoundError{}
+	ok := iter.Next()
+	if !ok {
+		err := iter.Err()
```

Comment:
Usually you'd write ```if ok := iter.Next(); !ok { ...```

Reply:
I prefer to keep the success path on the left, rather than nested in if blocks. A matter of style I guess 🤔